### PR TITLE
Modernize tests

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: pulp_ansible/tests/functional/api/collection/test_signatures.py [hunk 1]
```diff
@@ -188,16 +188,18 @@ def test_sync_signatures(
     monitor_task(sync_response.task)
     repo = ansible_repo_api_client.read(new_repo.pulp_href)
 
-    content_response = get_content(repo.to_dict())
-    assert len(content_response["ansible.collection_version"]) == 2
-    assert len(content_response["ansible.collection_signature"]) == 1
+    content_response = ansible_repo_version_api_client.read(repo.latest_version_href)
+    print(content_response)
```
Comment (gerrod3) [inline pulp_ansible/tests/functional/api/collection/test_signatures.py:192]:
```suggestion
```

File: pulp_ansible/tests/functional/api/collection/test_signatures.py [hunk 2]
```diff
@@ -215,6 +217,7 @@ def test_sync_signatures_only(
     monitor_task(sync_response.task)
     repo = ansible_repo_api_client.read(new_repo.pulp_href)
 
-    content_response = get_content(repo.to_dict())
-    assert len(content_response["ansible.collection_version"]) == 1
-    assert len(content_response["ansible.collection_signature"]) == 1
+    content_response = ansible_repo_version_api_client.read(repo.latest_version_href)
+    print(content_response)
```
Comment (gerrod3) [inline pulp_ansible/tests/functional/api/collection/test_signatures.py:221]:
```suggestion
```

**Review by gerrod3 — CHANGES_REQUESTED** [review]
How close are we now to fully removing pulp_smash?

File: setup.py [hunk 3]
```diff
@@ -18,20 +18,6 @@
     author_email="pulp-dev@redhat.com",
     url="https://github.com/pulp/pulp_ansible",
     install_requires=requirements,
-    extras_require={
-        "test": [
-            "coverage",
-            "dynaconf",
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply setup.py:24]: If, so this is bad design to have them defined in two places.

**Comment by mdellweg** [comment]
> How close are we now to fully removing pulp_smash?

There are still some tests to convert. Removing the v2 api & tests will be a big milestone. My impression is the last tests are also not the easiest...

File: setup.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mdellweg) [reply setup.py:24]: cc @pulp/galaxy

**Review by gerrod3 — APPROVED** [review]

**Pull request merged**